Tasks/JenkinsDownloadArtifacts/ArtifactDetails/ArtifactDetailsDownloader.ts

```diff
@@ -0,0 +1,125 @@
+import * as Q from 'q';
+import * as  tl from 'vsts-task-lib/task';
+
+import {JenkinsRestClient} from "./JenkinsRestClient"
+import {CommitsDownloader} from "./CommitsDownloader"
+import {WorkItemsDownloader} from "./WorkItemsDownloader"
+
+var handlebars = require('handlebars');
+
+export class ArtifactDetailsDownloader {
+    public DownloadCommitsAndWorkItems(): Q.Promise<void> {
+        let defer: Q.Deferred<any> = Q.defer<any>();
+
+        console.log(tl.loc("DownloadingCommitsAndWorkItems"));
+        
+        let startBuildId: number = parseInt(tl.getInput("startJenkinsBuildNumber"))
+        let commitsDownloader: CommitsDownloader = new CommitsDownloader();
+
+        if (!startBuildId) {
+            let jenkinsBuild = tl.getInput('jenkinsBuild', true);
+            let endBuildId: string = "";
+
+            if (jenkinsBuild === 'LastSuccessfulBuild') {
+                // If its LastSuccessfulBuild, we don't need to fetch the build number, we could just use the lastSuccessfulFull macro
+                endBuildId = 'lastSuccessfulBuild';
+            }
+            else {
+                // if its not LastSuccessfulBuild try to get the build number from input.
+                endBuildId = tl.getInput("jenkinsBuildNumber", false)
+            }
+
+            console.log(tl.loc("JenkinsDownloadingChangeFromCurrentBuild", endBuildId));
+
+            commitsDownloader.DownloadFromSingleBuildAndSave(endBuildId).then((commits: string) => {
+                    let commitMessages: string[] = CommitsDownloader.GetCommitMessagesFromCommits(commits);
+                    let workItemsDownloader: WorkItemsDownloader = new WorkItemsDownloader(commitMessages); 
+                    workItemsDownloader.DownloadFromSingleBuildAndSave(endBuildId).then(() => {
+                        defer.resolve(null);
+                    }, (error) => {
+                        defer.reject(error);
+                    });
+            }, (error) => {
+                defer.reject(error);
+            });
+        }
+        else {
+            let endBuildId = parseInt(tl.getInput("jenkinsBuildNumber", true));
```

Comment:
Do we always expect it to be number? Add type and handle NaN in parseInt.

Reply:
Fixed.